.github/workflows/cleanEnvironments.js

```diff
@@ -65,18 +89,82 @@ async function cleanEnvironment(environment, snapshot, token) {
       ? responseInJson.non_field_errors
       : responseInJson.__all__
   ) {
-    console.warn(
-      `${environment.name}: ${
-        responseInJson.non_field_errors
-          ? responseInJson.non_field_errors
-          : responseInJson.__all__
-      }`,
-    );
+    const warning = responseInJson.non_field_errors
+      ? responseInJson.non_field_errors
+      : responseInJson.__all__;
+    console.warn(`${environment.name}: ${warning}`);
+    sendWarningOnSlack = "true";
+    warningMessage += `Could not revert snapshot on ${environment.domain}: ${warning}.\n`;
   } else {
     await waitUntilTaskInProgress(responseInJson.task_id, environment.name);
   }
 }
 
+async function getSnapshotsForRestore(token) {
+  const snapshotsResponse = await fetch(
+    `${pathToCloudAPI}organizations/saleor/backups/`,
+    {
+      method: "GET",
+      headers: {
+        Authorization: `Token ${token}`,
+        Accept: "application/json",
+        "Content-Type": "application/json;charset=UTF-8",
+      },
+    },
+  );
+  const allSnapshots = await snapshotsResponse.json();
+  return allSnapshots.filter(snapshot => {
+    return snapshot.name.includes(snapshotName);
+  });
+}
+
+function sortSnapshots(snapshotList) {
+  // This function is used to sort snapshots by their version
+  // It returns sorted list of snapshots in descending order
+
+  return snapshotList.sort(function (a, b) {
+    return compareVersions(a.saleor_version, b.saleor_version);
+  });
+}
+
+function compareVersions(versionA, versionB) {
```

Comment:
What do you think about using `semver` for that? https://github.com/npm/node-semver (assuming the versions we are comparing are compliant with semver specs)